habitat/core/embodied_task.py

```diff
@@ -140,6 +230,117 @@ def __init__(
+            assert (
+                measure_type is not None
+            ), "invalid measurement type {}".format(measurement_cfg.TYPE)
+            print("measurement type {}".format(measurement_cfg.TYPE))
+            task_measurements.append(
+                measure_type(
+                    sim=sim, config=measurement_cfg, dataset=dataset, task=self
+                )
+            )
+        self.measurements = Measurements(task_measurements)
+
+        task_sensors = []
+        for sensor_name in config.SENSORS:
+            sensor_cfg = getattr(config, sensor_name)
+            sensor_type = registry.get_sensor(sensor_cfg.TYPE)
+            assert sensor_type is not None, "invalid sensor type {}".format(
+                sensor_cfg.TYPE
+            )
+            print("sensor type {}".format(sensor_cfg.TYPE))
+            task_sensors.append(
+                sensor_type(sim=sim, config=sensor_cfg, dataset=dataset)
+            )
+
+        self.sensor_suite = SensorSuite(task_sensors)
+
+        # TODO: move sensors and measures to `_init_entities`
+        self.actions = self._init_entities(
+            entity_names=config.POSSIBLE_ACTIONS,
+            register_func=registry.get_task_action,
+            entities_config=self._config.ACTIONS,
+        )
+
+    def _init_entities(
+        self, entity_names, register_func, entities_config=None
+    ):
+        if entities_config is None:
+            entities_config = self._config
+
+        entities = OrderedDict()
+        for entity_name in entity_names:
+            entity_cfg = getattr(entities_config, entity_name)
+            entity_type = register_func(entity_cfg.TYPE)
+            assert (
+                entity_type is not None
+            ), f"invalid {entity_name} type {entity_cfg.TYPE}"
+            entities[entity_name] = entity_type(
+                sim=self._sim,
+                config=entity_cfg,
+                dataset=self._dataset,
+                task=self,
+            )
+        return entities
+
+    def reset(self, episode):
+        observations = self._sim.reset()
+        observations.update(
+            self.sensor_suite.get_observations(
+                observations=observations, episode=episode
+            )
+        )
+
+        for action_instance in self.actions.values():
+            action_instance.reset(episode=episode)
+
+        return observations
+
+    def step(self, action: Union[int, Dict[str, Any]], episode: Type[Episode]):
+        if "action_args" not in action or action["action_args"] is None:
+            action["action_args"] = {}
+        action_name = action["action"]
+        if isinstance(action_name, (int, np.integer)):
+            if action_name >= len(self.actions):
+                raise ValueError(
+                    f"Action index '{action_name}' is out of range."
+                )
+            action_name = list(self.actions.keys())[action_name]
```

Comment:
Probably doesn't matter, but can we have a `self._action_keys = list(self.actions.keys())` in the `__init__` method and use that here?  Making this on every step seems excessive.

Reply:
Actually, shouldn't this be `action_name = self.get_action_name(action_name)`